File: v1/apps/scanner/internal/models/result.go
package models

type Severity string

const (
	SeverityCritical Severity = "critical"
	SeverityWarning  Severity = "warning"
	SeverityInfo     Severity = "info"
)

type CheckResult struct {
	ScanID      string   `json:"scanId"`
	Module      string   `json:"module"`
	Check       string   `json:"check"`
	Passed      bool     `json:"passed"`
	Severity    Severity `json:"severity"`
	Title       string   `json:"title"`
	Description string   `json:"description"`
	Fix         string   `json:"fix"`
}

File: v1/apps/web/models/result.ts
import mongoose, { Schema, Document, Model } from "mongoose";

export type Severity = "critical" | "warning" | "info";

export interface IResult extends Document {
  scanId: string;
  module: string;
  check: string;
  passed: boolean;
  severity: Severity;
  title: string;
  description: string;
  fix: string;
  completedAt: Date;
}

const ResultSchema = new Schema<IResult>({
  scanId: { type: String, required: true, index: true },
  module: { type: String, required: true },
  check: { type: String, required: true },
  passed: { type: Boolean, required: true },
  severity: {
    type: String,
    enum: ["critical", "warning", "info"],
    required: true,
  },
  title: { type: String, required: true },
  description: { type: String, default: "" },
  fix: { type: String, default: "" },
  completedAt: { type: Date, default: Date.now },
});

export const Result: Model<IResult> =
  mongoose.models.Result || mongoose.model<IResult>("Result", ResultSchema);